apps/application-system/api/src/app/modules/payment/payment.controller.ts

```diff
@@ -59,18 +61,39 @@ export class PaymentController {
   async createCharge(
     @CurrentUser() user: User,
     @Param('application_id', new ParseUUIDPipe()) applicationId: string,
+    @Body() payload: CreateChargeInput,
   ): Promise<CreatePaymentResponseDto> {
     if (!isUuid(applicationId)) {
       throw new BadRequestException(`ApplicationId is on wrong format.`)
     }
+    const allCatalogs = await this.paymentAPI.getCatalogByPerformingOrg(
```

Comment:
Maybe its better to call the API DOMAINS resolver instead of clients paymentAPI.... ?